server/tests/controllerTests/user.test.js

```diff
@@ -124,3 +123,99 @@ describe('GET api/v1/users', () => {
       .equals('Users found');
   });
 });
+
+describe('GET api/v1/users/moses', () => {
+  it('should return an error if user is not signed in', async () => {
+    const response = await chai.request(app)
+      .get('/api/v1/users/moses');
+
+    expect(response.status).to.be.equal(403);
+    expect(response.body.message).to.be.deep
+      .equals('You need to sign in first');
+  });
+
+  it('should return an error if token could not be verified', async () => {
+    const response = await chai.request(app)
+      .get('/api/v1/users/moses')
+      .set('x-access-token', fakeToken);
+
+    expect(response.status).to.be.equal(401);
+    expect(response.body.message).to.be.deep
+      .equals('Sorry, try signing in again');
+  });
+
+  it('should return error for user that do not exit', async () => {
+    const response = await chai.request(app)
+      .get('/api/v1/users/uwabuwa')
+      .set('x-access-token', token);
+
+    expect(response.status).to.be.equal(404);
+    expect(response.body.message).to.be.deep
+      .equals('User not found');
+  });
+
+  it('should return the user', async () => {
+    const response = await chai.request(app)
+      .get('/api/v1/users/i_amtheo')
+      .set('x-access-token', token);
+
+    expect(response.status).to.be.equal(200);
+    expect(response.body.message).to.be.deep
+      .equals('User found');
+  });
+});
+
+describe('UPDATE api/v1/users/:username', () => {
+  it('should return an error if user is not signed in', async () => {
+    const response = await chai.request(app)
+      .put('/api/v1/users/i_amsull')
+      .send(goodUserUpdate);
+
+    expect(response.status).to.be.equal(403);
+    expect(response.body.message).to.be.deep
+      .equals('You need to sign in first');
+  });
+
+  it('should return an error if token could not be verified', async () => {
```

Comment:
Is this the same with the one above? I see an it block with the same description.